### PR TITLE
Bug fix create primitive net name

### PR DESCRIPTION
**Pull request by ring630**

### CODE REVIEW

**Comment by MaxJPRey** [comment]
@ring630 We might have to add a deprecation warning here because we are modifying the signature of an existing method.

File: pyaedt/modeler/Primitives3DLayout.py [hunk 1]
```diff
@@ -736,7 +737,7 @@ def create_via(
         return name
 
     @pyaedt_function_handler()
-    def create_circle(self, layername, x, y, radius, name=None, netname=None):
+    def create_circle(self, layername, x, y, radius, name=None, net_name=None):
```
Comment (maxcapodi78) [inline pyaedt/modeler/Primitives3DLayout.py:740]:
```suggestion
    def create_circle(self, layername, x, y, radius, name=None, net_name=None, **kwargs):
```

File: pyaedt/modeler/Primitives3DLayout.py [hunk 2]
```diff
@@ -766,6 +767,10 @@ def create_circle(self, layername, x, y, radius, name=None, netname=None):
 
         >>> oEditor.CreateCircle
         """
+        warnings.warn(
+            "`netname` is deprecated. Use `net_name` instead.",
+            DeprecationWarning,
+        )
```
Comment (maxcapodi78) [inline pyaedt/modeler/Primitives3DLayout.py:773]:
```suggestion
        if "netname" in kwargs:
            warnings.warn(
                "`netname` is deprecated. Use `net_name` instead.",
                DeprecationWarning,
            )
```

**Review by maxcapodi78 — COMMENTED** [review]
Few suggestion to keep the back compatibility

File: pyaedt/modeler/Primitives3DLayout.py [hunk 3]
```diff
@@ -783,28 +788,33 @@ def create_circle(self, layername, x, y, radius, name=None, netname=None):
...
+        self._circles[name] = primitive
+
+        if net_name:
+            primitive.change_property(property_val=["NAME:Net", "Value:=", net_name])
+
         return name
 
     @pyaedt_function_handler()
-    def create_rectangle(self, layername, origin, dimensions, corner_radius=0, angle=0, name=None, netname=None):
+    def create_rectangle(self, layername, origin, dimensions, corner_radius=0, angle=0, name=None, net_name=None):
```
Comment (maxcapodi78) [inline pyaedt/modeler/Primitives3DLayout.py:800]:
```suggestion
    def create_rectangle(self, layername, origin, dimensions, corner_radius=0, angle=0, name=None, net_name=None, **kwargs):
```

File: pyaedt/modeler/Primitives3DLayout.py [hunk 4]
```diff
@@ -818,7 +828,10 @@ def create_rectangle(self, layername, origin, dimensions, corner_radius=0, angle
 
         >>> oEditor.CreateRectangle
         """
-
+        warnings.warn(
+            "`netname` is deprecated. Use `net_name` instead.",
+            DeprecationWarning,
+        )
```
Comment (maxcapodi78) [inline pyaedt/modeler/Primitives3DLayout.py:834]:
```suggestion
        if "netname" in kwargs:
            warnings.warn(
                "`netname` is deprecated. Use `net_name` instead.",
                DeprecationWarning,
            )
            net_name=kwargs["netname"]
```

File: pyaedt/modeler/Primitives3DLayout.py [hunk 5]
```diff
@@ -839,19 +852,24 @@ def create_rectangle(self, layername, origin, dimensions, corner_radius=0, angle
...
+        self._rectangles[name] = primitive
+
+        if net_name:
+            primitive.change_property(property_val=["NAME:Net", "Value:=", net_name])
+
         return name
 
     @pyaedt_function_handler()
-    def create_line(self, layername, center_line_list, lw=1, start_style=0, end_style=0, name=None, netname=None):
+    def create_line(self, layername, center_line_list, lw=1, start_style=0, end_style=0, name=None, net_name=None):
```
Comment (maxcapodi78) [inline pyaedt/modeler/Primitives3DLayout.py:864]:
```suggestion
    def create_line(self, layername, center_line_list, lw=1, start_style=0, end_style=0, name=None, net_name=None, **kwargs):
```

File: pyaedt/modeler/Primitives3DLayout.py [hunk 6]
```diff
@@ -882,6 +900,10 @@ def create_line(self, layername, center_line_list, lw=1, start_style=0, end_styl
 
         >>> oEditor.CreateLine
         """
+        warnings.warn(
+            "`netname` is deprecated. Use `net_name` instead.",
+            DeprecationWarning,
+        )
```
Comment (maxcapodi78) [inline pyaedt/modeler/Primitives3DLayout.py:906]:
```suggestion
        if "netname" in kwargs:
            warnings.warn(
                "`netname` is deprecated. Use `net_name` instead.",
                DeprecationWarning,
            )
            net_name=kwargs["netname"]
```

**Review by maxcapodi78 — COMMENTED** [review]
added back compatibility

**Review by maxcapodi78 — APPROVED** [review]
LGTM

**Pull request merged**